lib/kafka/cluster.rb

```diff
@@ -46,6 +46,10 @@ def add_target_topics(topics)
         @target_topics.merge(new_topics)
 
         refresh_metadata!
+
+        cluster_info.topics.each do |topic|
+          Protocol.handle_error(topic.topic_error_code)
+        end
```

Comment:
I think this should either go in `refresh_metadata!` or only check the topics that are being added.